crawl-ref/source/actor.cc

```diff
@@ -1032,6 +1033,107 @@ void actor::collide(coord_def newpos, const actor *agent, int pow)
     }
 }
 
+/**
+ * @brief Attempts to knock this actor away from a specific tile,
+ *        using repeated one-tile knockbacks. Any LOS requirements
+ *        must be checked by the calling function.
+ * @param cause The actor responsible for the knockback.
```

Comment:
Pass `cause` as a reference (&) not as a pointer (*). It's always non-null, so we don't need to invite bugs by allowing it to be nullable.